src/stores/models/Oracle.test.js

```diff
@@ -0,0 +1,206 @@
+import { observable } from 'mobx';
+import { Token, Phases, OracleStatus } from 'constants';
+import Oracle from './Oracle';
+import { decimalToSatoshi } from '../../helpers/utility';
+
+const localStorageMock = {
+  getItem: jest.fn(),
+  setItem: jest.fn(),
+  clear: jest.fn(),
+};
+const navigatorMock = {};
+global.navigator = navigatorMock;
+global.localStorage = localStorageMock;
+
+const { QTUM, BOT } = Token;
+const { BETTING, VOTING, RESULT_SETTING } = Phases;
+
+describe('models/Oracle', () => {
+  let oracle;
+  const addr = {
+    address: 'qSu4uU8MGp2Ya6j9kQZAtizUfC82aCvGT1',
+    qtum: 2000,
+    bot: 100,
+  };
+  const app = observable({
+    sortBy: 'ASC',
+    wallet: { addresses: [addr] },
+    global: { syncBlockNum: 0 },
+    refreshing: false,
+    ui: { location: 0 },
+  }); // mock the appstore
+  beforeEach(() => {
+    oracle = null;
+  });
+
+  /** all following test cases target specifically to the mock backend, eg. network/graphql/__mocks__/queries.js */
+  it('Constructor confirmed betting', async () => {
+    const input = observable({
+      txid: '0',
+      amounts: [decimalToSatoshi('99'), decimalToSatoshi('100')],
+      consensusThreshold: decimalToSatoshi('100'),
+      address: '02e91962156da21fae38e65038279c020347e4ff',
+      topicAddress: '4044f951857f2885d66d29a475235dacdaddea84',
+      resultSetEndTime: '10',
+      endTime: '20',
+      options: ['1', '2'],
+      token: QTUM,
+      optionIdxs: [0, 1],
+      status: OracleStatus.VOTING,
+      hashId: 'hahaha',
+    });
+    oracle = new Oracle(input, app);
+
+    expect(oracle.app).toBe(app);
+    expect(oracle.amounts.length).toBe(input.amounts.length);
+    expect(oracle.amounts[0]).toBe(99);
+    expect(oracle.amounts[1]).toBe(100);
+    expect(oracle.consensusThreshold).toBe(100);
+    expect(oracle.url).toBe('/oracle/4044f951857f2885d66d29a475235dacdaddea84/02e91962156da21fae38e65038279c020347e4ff/0');
+    expect(oracle.endTime).toBe('20');
+    expect(oracle.options.length).toBe(input.options.length);
+
+    expect(oracle.isUpcoming).not.toBeTruthy();
+    expect(oracle.unconfirmed).not.toBeTruthy();
+    expect(oracle.isPending).not.toBeTruthy();
+    expect(oracle.isArchived).not.toBeTruthy();
+    expect(oracle.isOpenResultSetting).not.toBeTruthy();
+    expect(oracle.phase).toBe(BETTING);
+  });
+
+  it('Constructor unconfirmed result setting', async () => {
```

Comment:
dont think there should ever be an unconfirmed Result Setting Oracle. only an unconfirmed Betting Oracle which is created when you create an event.

Reply:
hahaha I thought it would be funny if this case fails, I wanted to double check the model is iron-safe.
It's just another dumb case, again I can remove if you say so.

Reply:
Yes please remove, not possible to reach.